core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/BulkImport.java

```diff
@@ -576,7 +587,12 @@ private Text toText(byte[] row) {
         Thread.currentThread().interrupt();
         throw new RuntimeException(e);
       } catch (ExecutionException e) {
-        throw new RuntimeException(e);
+        // clean up exception for user
+        Throwable t = e.getCause();
+        if (t instanceof IllegalArgumentException) {
+          throw (IllegalArgumentException) t;
+        } else
+          throw new RuntimeException(t);
       }
```

Comment:
I would wrap `e` so that stack traces are not lost and anyone getting the exceptions can trace the full code path from their code to the background thread.

```suggestion
        Throwable t = e.getCause();
        if (t instanceof IllegalArgumentException) {
          throw new IllegalArgumentException(e);
        } else
          throw new RuntimeException(e);
      }
```

Reply:
I am not sure this is better than just the Runtime.  I think this will wrap an IllegalArgument around the whole stack, which already has an IllegalArgument at the bottom.  I will play around.

Reply:
See my other comment.  You OK with me keeping the exception handling as-is?

Reply:
Sorry for the delay @milleruntime.  This fell off my radar.